astropy/units/format/console.py

```diff
@@ -67,42 +67,39 @@ def format_exponential_notation(cls, val):
 
     @classmethod
     def to_string(cls, unit, inline=True):
-        if isinstance(unit, core.CompositeUnit):
-            if unit.scale == 1:
-                s = ""
+        if unit.scale == 1:
+            s = ""
+        else:
+            s = cls.format_exponential_notation(unit.scale)
+
+        if len(unit.bases):
```

Comment:
I tried ``bool(len(u.m.bases))`` and this evaluates to True. When is ``len(unit.bases)`` False?

Reply:
Only for dimensionless - which can have a scale:
```
u.percent.decompose()
# Unit(dimensionless with a scale of 0.01)
f"{u.percent.decompose()}"
# 0.01
```